packages/react/src/class-names/index.ts

```diff
@@ -3,11 +3,18 @@ import type { ReactNode, CSSProperties } from 'react';
 import type { BasicTemplateInterpolations, CssFunction } from '../types';
 import { createSetupError } from '../utils/error';
 
-export type Interpolations = (BasicTemplateInterpolations | CssFunction | CssFunction[])[];
+export type Interpolations<TProps> = (
+  | BasicTemplateInterpolations
+  | CssFunction<TProps>
+  | CssFunction<TProps>[]
+)[];
 
-export interface ClassNamesProps {
+export interface ClassNamesProps<TProps> {
   children: (opts: {
-    css: (css: CssFunction | CssFunction[], ...interpolations: Interpolations) => string;
+    css: (
```

Comment:
Can this be an interface like in `styled` so that you separate the tagged template usage from the object call expression?

Or could you utilise the types from `css` instead? Then just change the return type

Reply:
yeah the CSS types approach seems pretty elegant, will try that